### ISSUE TITLE
ERR_UNSUPPORTED_API_LEVEL: Unsupported API level: 32

### ISSUE THREAD

**Issue by SahilSeta** [ionic-team/native-run#219]
**Version:**

native-run failed with error
ERR_UNSUPPORTED_API_LEVEL: Unsupported API level: 32

**Description:**
ERROR OCCUR WHILE USING "npx cap run android"

**Command Output:**

**SDK Info:**

**Comment by rflpsz** [comment]
I had the same problem, there are some fix for it?

**Comment by aliexalter** [comment]
How to downgrade to API level 31 as native-run only supports maximum API level 31.

**Comment by entrifis** [comment]
Just started my first app today, following the official documentation, and got the same error. Downgrading the API is giving another error about not having the right API installed

**Comment by pattodiablo** [comment]
same to me, what to do?

**Comment by aliexalter** [comment]
Try this
From Android Studio Install API level 31 from SDK manager and also install API 31 emulator from AVD manager.
Then use npx cap copy android or sync. Run project in android studio.
For live reload in cordova use API level 31 virtual device as target.

**Comment by inorganik** [comment]
same issue

**Comment by gpetrov** [comment]
It will be nice if capacitor supports Android api 32 as this is the default when installing Android Studio now.

Also Capacitor seems to be running all fine in the emulator with Android 32. So it is probably just the check.

**Comment by jcesarmobile** [comment]
SDK 32 is not even official yet, so not sure why Android Studio installs it instead of the latest stable (31) that was released just a few months ago.

We are not adding support for SDK 32 until it's official and stable, please, downgrade to SDK 30 or 31.

**Comment by inorganik** [comment]
Thanks for the info @jcesarmobile. Can you provide instructions for how to downgrade? I installed Android Studio recently and it shipped with SDK 32. I'm trying to publish an android version and keep hitting this error. Thanks

**Comment by jcesarmobile** [comment]
From Android Studio click the grey box with a blue arrow on the top right
Then uncheck all the SDK 32 related packages and check at least SDK platform tools and one of the Google Play system images for SDK 30 or SDK 31, see screenshot


![Captura de pantalla 2022-01-19 a las 15 10 02](https://user-images.githubusercontent.com/1637892/150147201-2437431b-12dd-4d96-a903-0a387a67987b.png)

**Comment by OgeHub** [comment]
@jcesarmobile I downloaded Android Studio 25th January 2022, and had this issue of unsupported API level. Then I followed the steps your suggested to install Android 12 and 11 with lower API level. But I am now getting error of unsuitable API installation. Please I need help. Below is screenshots of my SDK installations and the error on my terminal.
![error1](https://user-images.githubusercontent.com/75125594/151365676-9028220b-e9ff-4d7a-9037-ceaf3a1e675b.png)
![sdk1](https://user-images.githubusercontent.com/75125594/151365691-b43b5a40-70bb-4cdf-94d5-e0058f720f7b.png)

**Comment by jcesarmobile** [comment]
check the "show package details" checkbox at the bottom, so you can see which components you downloaded, you need at least SDK platform tools and one of the Google Play system images

**Comment by OgeHub** [comment]
Hello @jcesarmobile , I have installed all the required packages as indicated in the screenshot you shared, but when I try to run my project, using npx cap run android, it gives me the error on this screenshot. I sorted for solution online; downloaded Java JDK, updated my environment variable and path, and restarted my vscode. I couldn't still solve the issue. Please I need your help on this. Thank you.
![gradle err](https://user-images.githubusercontent.com/75125594/151781926-928ffa13-a574-49ca-a31b-2d7db7925845.png)

**Comment by abiligiri** [comment]
> Hello @jcesarmobile , I have installed all the required packages as indicated in the screenshot you shared, but when I try to run my project, using npx cap run android, it gives me the error on this screenshot. I sorted for solution online; downloaded Java JDK, updated my environment variable and path, and restarted my vscode. I couldn't still solve the issue. Please I need your help on this. Thank you.
> ![gradle err](https://user-images.githubusercontent.com/75125594/151781926-928ffa13-a574-49ca-a31b-2d7db7925845.png)

export JAVA_HOME=/Applications/Android\ Studio.app/Contents/jre/Contents/Home on macOS, you should do the equivalent on Windows

**Comment by cyantree** [comment]
Maybe it's useful for someone. Most of the info has already been posted by other users.
I installed Android Studio via portable ZIP file on Windows and needed to do the following things to get it running:

In the shell I want to use for building/running:
```
export CAPACITOR_ANDROID_STUDIO_PATH="D:/Apps/Android Studio/bin/studio64.exe"
export ANDROID_SDK_ROOT=D:/Apps/Android-SDK
export JAVA_HOME="D:/Apps/Android Studio/jre"
```

Android Studio Chipmunk came with a bundled Android API 33 so `yarn cap run android` failed with the following error: `ERR_UNSUPPORTED_API_LEVEL: Unsupported API level: 33`

Running `yarn native-run android --sdk-info` gave me some more info about what was missing for Android API 32 which I already installed:
```
(!) Missing Packages: Google Play Intel x86 Atom System Image  system-images;android-32;google_apis_playstore;x86  /.+/
```

So I installed it showing the package details in Android Studio and selecting the requested image.

Afterwards I could run the app with both Android API 33 and 32 installed. :)
